### ISSUE TITLE
Should `iinit` show default values or available choices?

### ISSUE THREAD

**Issue by alanking** [irods/irods#7142]
## Feature

We are adding more prompts to `iinit` to allow for specifying SSL and non-native authentication schemes: https://github.com/irods/irods_client_icommands/pull/338

Here are the options for these prompts:

**1. No hints in the prompts**

This would align roughly with what we have now. Here is what this would look like:
```bash
$ iinit --with-ssl --prompt-auth-scheme
 ERROR: environment_properties::capture: missing environment file. should be at [/root/.irods/irods_environment.json]
Enter the host name (DNS) of the server to connect to: localhost
Enter the port number: 1247
Enter your iRODS user name: alan
Enter your iRODS zone: tempZone
Enter the server verification level: hostname
...
Enter server authentication scheme: native
...
```

**2. Default values in the prompts (i.e. what happens when you enter an empty value)**
This would show default values based on historical `iinit` behavior as well as any values set in the environment and/or existing active client environment file. Here's what this could look like with no existing client environment settings:
```bash
$ iinit --with-ssl --prompt-auth-scheme
 ERROR: environment_properties::capture: missing environment file. should be at [/root/.irods/irods_environment.json]
Enter the host name (DNS) of the server to connect to: localhost
Enter the port number: 1247
Enter your iRODS user name: alan
Enter your iRODS zone: tempZone
Enter the server verification level [hostname]:
...
Enter server authentication scheme [native]:
...
```

Here's what this could look like with some existing client environment settings, but wishing to change something via `iinit`:
```bash
$ iinit --with-ssl --prompt-auth-scheme
 ERROR: environment_properties::capture: missing environment file. should be at [/root/.irods/irods_environment.json]
Enter the host name (DNS) of the server to connect to [icat.example.org]:
Enter the port number [1247]: 1247
Enter your iRODS user name [alan]:
Enter your iRODS zone [tempZone]:
Enter the server verification level [hostname]:
...
Enter server authentication scheme [native]:
...
```

**3. Show all possible values, where applicable.**
This would present all of the (known) valid options to the user. See server verification level and auth scheme prompts:
```bash
$ iinit --with-ssl --prompt-auth-scheme
 ERROR: environment_properties::capture: missing environment file. should be at [/root/.irods/irods_environment.json]
Enter the host name (DNS) of the server to connect to: localhost
Enter the port number: 1247
Enter your iRODS user name: alan
Enter your iRODS zone: tempZone
Enter the server verification level [cert, hostname, verify]:
...
Enter server authentication scheme [native, pam_password]:
...
```
Would this also include some indication of what the default value would be?

This may depend on or at least be related to this issue: https://github.com/irods/irods/issues/6752

---

From this discussion: https://github.com/irods/irods_client_icommands/pull/338#discussion_r1048798877

**Comment by korydraughn** [comment]
Option 3 seems nice, but I'm concerned that reporting all available options can help malicious users.

Option 2 seems the safest path forward, while also providing helpful info about the user's existing configuration.

I vote for option 2 for now.

**Comment by trel** [comment]
When there is a very tight 'enum'... then 3 could work, but that also makes it harder to use 'older' iCommands with newer servers if we change the enum...

I think 2 is the right mix as well.

**Comment by alanking** [comment]
irods/irods_client_icommands#338 has been merged and it implements option 2.

If we want to implement option 3, we will need to implement #6752. As such, I think that this issue can be closed as resolved/invalid due to the completion of #6398 and #3164.

Thoughts?
